backend/endpoints/dependency_factory.go

```diff
@@ -80,3 +82,8 @@ func getDependency[T any](factory DependencyFactory) T {
 
 	return factory.GetDependency(factory.GetDepFromType(typeMapping)).(T)
 }
+
+func setFrontEndID(id int, df interface{}) {
+	stype := reflect.ValueOf(df).Elem()
+	stype.FieldByName("FrontEndID").SetInt(int64(id))
+}
```

Comment:
1. What is this for?
2. Please don't use reflection where its not needed
3. This only works on the very specific implementation of DF which is our `dependencyProvider`, it will panic on everything else